core/commands/public_keys_command.rb

```diff
@@ -78,29 +90,23 @@ def init
   # Connect and add ssh key on server
   # @param machine [Hash] information about machine to connect
   def configure_server_ssh_key(machine)
-    exit_code = SUCCESS_RESULT
-    options = Net::SSH.configuration_for(machine['network'], true)
-    options[:auth_methods] = %w[publickey none]
-    options[:verify_host_key] = false
-    options[:keys] = [machine['keyfile']]
-    begin
-      Net::SSH.start(machine['network'], machine['whoami'], options) do |ssh|
-        add_key(ssh)
-      end
-    rescue StandardError
-      @ui.error("Could not initiate connection to the node '#{machine['name']}'")
-      exit_code = ERROR_RESULT
+    if add_key(machine).error?
```

Comment:
You do not need to wrap the Result object with the ERROR_RESULT and SUCCESS_RESULT as they are already instances of the Result object.